### PR TITLE
new schema for data in deployment

### PR DESCRIPTION
**Pull request by konstan**
Related to https://github.com/nuvla/api-server/issues/325

### CODE REVIEW

File: code/src/sixsq/nuvla/server/resources/spec/deployment.cljc [hunk 1]
```diff
@@ -148,28 +204,13 @@
...
-
-
-
 (def deployment-keys-spec
   (su/merge-keys-specs [common/common-attrs
                         {:req-un [::module
                                   ::state
                                   ::api-endpoint]
                          :opt-un [::api-credentials
-                                  ::data-objects            ;;deprecated
```
Comment (konstan) [inline code/src/sixsq/nuvla/server/resources/spec/deployment.cljc]: @0xbase12 Do I have to put them back with full definitions behind?

Comment (0xbase12) [reply code/src/sixsq/nuvla/server/resources/spec/deployment.cljc]: Yes, if not, old deployment documents will fail spec on edit.

Comment (0xbase12) [reply code/src/sixsq/nuvla/server/resources/spec/deployment.cljc]: If you want, you can create an ES migration script (to be executed on upgrade of system) to update existing document to the new schema.

Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/spec/deployment.cljc]: I put back the removed fields. Also created https://github.com/nuvla/api-server/issues/333

**Pull request merged**